File: php/features.php
<?php
declare(strict_types=1);

// RbfVpn SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RbfVpnFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RbfVpnBaseFeature();
            case "test":
                return new RbfVpnTestFeature();
            default:
                return new RbfVpnBaseFeature();
        }
    }
}

File: py/features.py
# RbfVpn SDK feature factory

from feature.base_feature import RbfVpnBaseFeature
from feature.test_feature import RbfVpnTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RbfVpnBaseFeature(),
        "test": lambda: RbfVpnTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
